src/sources/amqp.rs

```diff
@@ -234,25 +234,20 @@ fn populate_event(
     // This handles the transition from the original timestamp logic. Originally the
     // `timestamp_key` was populated by the `properties.timestamp()` time on the message, falling
     // back to calling `now()`.
-    match (log_namespace, timestamp) {
-        (LogNamespace::Vector, None) => {
-            log.insert(metadata_path!("vector", "ingest_timestamp"), Utc::now());
-        }
-        (LogNamespace::Vector, Some(timestamp)) => {
-            log.insert(
-                metadata_path!(AmqpSourceConfig::NAME, "timestamp"),
-                timestamp,
-            );
+    match log_namespace {
+        LogNamespace::Vector => {
+            if let Some(timestamp) = timestamp {
+                log.insert(
+                    metadata_path!(AmqpSourceConfig::NAME, "timestamp"),
+                    timestamp,
+                );
+            };
 
             log.insert(metadata_path!("vector", "ingest_timestamp"), Utc::now());
         }
-        (LogNamespace::Legacy, None) => {
-            log.try_insert(
-                (PathPrefix::Event, log_schema().timestamp_key()),
-                Utc::now(),
-            );
-        }
-        (LogNamespace::Legacy, Some(timestamp)) => {
+        LogNamespace::Legacy => {
+            timestamp.unwrap_or_else(Utc::now);
```

Comment:
looks like the result of this call needs to be assigned back to `let timestamp = `
, same for the other file.

Reply:
Resolved in https://github.com/vectordotdev/vector/pull/15205/commits/0cd444023c24375a2790253098d85281a258fc22